x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java

```diff
@@ -705,9 +708,17 @@ private CloseableHttpAsyncClient createHttpClient() {
                     .setConnectionRequestTimeout(Math.toIntExact(realmConfig.getSetting(HTTP_CONNECTION_READ_TIMEOUT).getSeconds()))
                     .setSocketTimeout(Math.toIntExact(realmConfig.getSetting(HTTP_SOCKET_TIMEOUT).getMillis()))
                     .build();
+                final TimeValue connectionTtl = realmConfig.getSetting(HTTP_CONNECTION_POOL_TTL);
                 HttpAsyncClientBuilder httpAsyncClientBuilder = HttpAsyncClients.custom()
                     .setConnectionManager(connectionManager)
-                    .setDefaultRequestConfig(requestConfig);
+                    .setDefaultRequestConfig(requestConfig)
+                    .setKeepAliveStrategy((response, context) -> {
+                        if (TimeValue.MINUS_ONE.equals(connectionTtl)) {
+                            return DefaultConnectionKeepAliveStrategy.INSTANCE.getKeepAliveDuration(response, context);
+                        } else {
+                            return connectionTtl.millis();
+                        }
```

Comment:
Should we completely ignore the server's keep alive if a TTL is configured?
I wonder if we want:
```suggestion
                        var serverKeepAlive = DefaultConnectionKeepAliveStrategy.INSTANCE.getKeepAliveDuration(response, context);
                        if (TimeValue.MINUS_ONE.equals(connectionTtl)) {
                            return serverKeepAlive
                        } else if (serverKeepAlive == -1) {
                            return connectionTtl.millis();
                        } else {
                            return Math.min(serverKeepAlive, connectionTtl.millis());
                        }
```

Reply:
Taking the minimum of the two makes sense. Thanks